crates/modelardb_server/src/common_test.rs

```diff
@@ -0,0 +1,219 @@
+    let object_store_url = storage::QUERY_DATA_FOLDER_SCHEME_WITH_HOST
+        .try_into()
+        .unwrap();
+    let object_store = Arc::new(LocalFileSystem::new_with_prefix(path).unwrap());
+    session
+        .runtime_env()
+        .register_object_store(&object_store_url, object_store);
+
+    let storage_engine = RwLock::new(
+        StorageEngine::try_new(path.to_owned(), None, metadata_manager.clone(), true)
+            .await
+            .unwrap(),
+    );
+
+    Arc::new(Context {
+        metadata_manager,
+        session,
+        storage_engine,
+    })
+}
+
+/// Return a [`MetadataManager`] with 5 MiBs reserved for uncompressed data, 5 MiBs reserved for
+/// compressed data, and the data folder set to `path`. Reducing the amount of reserved memory
+/// makes it faster to run unit tests.
+pub fn test_metadata_manager(path: &Path) -> MetadataManager {
+    let mut metadata_manager = MetadataManager::try_new(path, ServerMode::Edge).unwrap();
+
+    metadata_manager.uncompressed_reserved_memory_in_bytes = 5 * 1024 * 1024; // 5 MiB
+    metadata_manager.compressed_reserved_memory_in_bytes = 5 * 1024 * 1024; // 5 MiB
+
+    metadata_manager
+}
+
+/// Return a [`SessionContext`] without any additional optimizer rules.
+pub fn test_session_context() -> SessionContext {
+    let config = SessionConfig::new();
+    let runtime = Arc::new(RuntimeEnv::default());
+    let mut state = SessionState::with_config_rt(config, runtime);
+    for physical_optimizer_rule in optimizer::physical_optimizer_rules() {
+        state = state.add_physical_optimizer_rule(physical_optimizer_rule);
+    }
+    SessionContext::with_state(state)
+}
+
+/// Return [`ModelTableMetadata`] for a model table with a schema containing a tag column, a
+/// timestamp column, and two field columns.
+pub fn model_table_metadata() -> ModelTableMetadata {
+    let query_schema = Arc::new(Schema::new(vec![
+        Field::new("timestamp", ArrowTimestamp::DATA_TYPE, false),
+        Field::new("field_1", ArrowValue::DATA_TYPE, false),
+        Field::new("field_2", ArrowValue::DATA_TYPE, false),
+        Field::new("tag", DataType::Utf8, false),
+    ]));
+
+    let error_bounds = vec![
+        ErrorBound::try_new(0.0).unwrap(),
+        ErrorBound::try_new(1.0).unwrap(),
+        ErrorBound::try_new(5.0).unwrap(),
+        ErrorBound::try_new(0.0).unwrap(),
+    ];
+
+    let generated_columns = vec![None, None, None, None];
+
+    ModelTableMetadata::try_new(
+        "model_table".to_owned(),
+        query_schema,
+        error_bounds,
+        generated_columns,
+    )
+    .unwrap()
+}
+
+/// Return a [`RecordBatch`] containing three compressed segments.
+pub fn compressed_segments_record_batch() -> RecordBatch {
+    compressed_segments_record_batch_with_time(0, 0.0)
+}
+
+/// Return a [`RecordBatch`] containing three compressed segments. The compressed segments time
+/// range is from `time_ms` to `time_ms` + 3, while the value range is from `offset` + 5.2 to
```

Comment:
is there any specific reason for `5.2` and `34.2` being the offset?

Reply:
I do not know, these tests seems to have been added when #18 was merged as 70b785a but I cannot recall why these values where selected.